harmonica/tests/test_legendre.py

```diff
@@ -0,0 +1,203 @@
+    dp[3, 0] = (3 * sin - 15 * cos**2 * sin) / 2
+    dp[3, 1] = 3 / 2 * (5 * cos**3 - 10 * cos * sin**2 - cos)
+    dp[3, 2] = 30 * cos**2 * sin - 15 * sin**3
+    dp[3, 3] = 45 * sin**2 * cos
+    dp[4, 0] = (15 * cos - 35 * cos**3) * sin / 2
+    dp[4, 1] = (
+        35 * cos**4 - 15 * cos**2 + 15 * sin**2 - 105 * cos**2 * sin**2
+    ) / 2
+    dp[4, 2] = 105 * sin * cos**3 - 105 * cos * sin**3 - 15 * sin * cos
+    dp[4, 3] = 315 * cos**2 * sin**2 - 105 * sin**4
+    dp[4, 4] = 420 * sin**3 * cos
+    return dp
+
+
+def schmidt_normalization(p):
+    "Calculate the Schmidt normalization factor"
+    max_degree = p.shape[0] - 1
+    for n in range(max_degree + 1):
+        for m in range(n + 1):
+            if m > 0:
+                p[n, m] *= np.sqrt(2 * math.factorial(n - m) / math.factorial(n + m))
+
+
+def full_normalization(p):
+    "Calculate the full normalization factor"
+    max_degree = p.shape[0] - 1
+    for n in range(max_degree + 1):
+        for m in range(n + 1):
+            if m == 0:
+                p[n, m] *= np.sqrt(2 * n + 1)
+            else:
+                p[n, m] *= np.sqrt(
+                    2 * (2 * n + 1) * math.factorial(n - m) / math.factorial(n + m)
+                )
+
+
+@pytest.mark.parametrize(
+    "func,norm",
+    (
+        (associated_legendre, None),
+        (associated_legendre_schmidt, schmidt_normalization),
+        (associated_legendre_full, full_normalization),
+    ),
+    ids=["unnormalized", "schmidt", "full"],
+)
+def test_associated_legendre_function_analytical(func, norm):
+    "Check if the first few degrees match analytical expressions"
+    for angle in np.linspace(0, np.pi, 360):
+        x = np.cos(angle)
+        # Analytical expression
+        p_analytical = legendre_analytical(x)
+        max_degree = p_analytical.shape[0] - 1
+        if norm is not None:
+            norm(p_analytical)
+        # Numerical calculation
+        p = np.empty((max_degree + 1, max_degree + 1))
+        func(x, max_degree, p)
+        for n in range(max_degree + 1):
+            for m in range(n + 1):
+                np.testing.assert_allclose(p_analytical[n, m], p[n, m], atol=1e-10)
+
+
+@pytest.mark.parametrize(
+    "func,deriv,norm",
+    (
+        (associated_legendre, associated_legendre_derivative, None),
+        (
+            associated_legendre_schmidt,
+            associated_legendre_schmidt_derivative,
+            schmidt_normalization,
+        ),
+        (
+            associated_legendre_full,
+            associated_legendre_full_derivative,
+            full_normalization,
+        ),
+    ),
+    ids=["unnormalized", "schmidt", "full"],
+)
+def test_associated_legendre_function_derivative_analytical(func, norm, deriv):
+    "Check if the first few degrees match analytical expressions"
+    for angle in np.linspace(0, np.pi, 360):
+        x = np.cos(angle)
+        # Analytical expression
+        dp_analytical = legendre_derivativeative_analytical(x)
```

Comment:
And also here, where the function is being used.
```suggestion
        dp_analytical = legendre_derivative_analytical(x)
```